homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -156,4 +162,11 @@ async def async_update(self) -> None:
             if not self._remaining_time or self._remaining_time.total_seconds() < 0:
                 await self._opendata.async_get_data()
         except OpendataTransportError:
-            _LOGGER.error("Unable to retrieve data from transport.opendata.ch")
+            self._connected = False
+            _LOGGER.warning(
+                "Unable to connect and retrieve data from transport.opendata.ch"
+            )
+        else:
+            if not self._connected:
+                self._connected = True
+                _LOGGER.info("Connection established with transport.opendata.ch")
```

Comment:
```suggestion
            self._attr_available = False
            _LOGGER.warning(
                "Unable to connect and retrieve data from transport.opendata.ch"
            )
        else:
            if not self._attr_available:
                self._attr_available = True
                _LOGGER.info("Connection established with transport.opendata.ch")
```

This should be enough. You by the way don't need this change if you're going to switch to a coordinator

Reply:
Ok, but one step after the other on my learning curve :)